forge-game/src/main/java/forge/game/player/Player.java

```diff
@@ -863,7 +863,20 @@ public void addCounterInternal(final CounterType counterType, final int n, final
 
         final int oldValue = getCounters(counterType);
         final int newValue = addAmount + oldValue;
-        this.setCounters(counterType, newValue, fireEvents);
+        this.setCounters(counterType, newValue, source, fireEvents);
+
+        if (counterType.is(CounterEnumType.RAD)) {
+            if (newValue > 0) {
+                String setCode = null;
+                if (params.containsKey(AbilityKey.Cause)) {
+                    SpellAbility cause = (SpellAbility) params.get(AbilityKey.Cause);
+                    setCode = cause.getHostCard().getSetCode();
+                }
+                createRadiationEffect(setCode);
+            } else {
+                removeRadiationEffect();
```

Comment:
you need to move this somewhere else
currently the effect stays around after you're cured